src/providers/mod.rs

```diff
@@ -72,20 +72,26 @@ pub trait MetadataProvider {
     }
 
     fn write_ssh_keys(&self, ssh_keys_user: String) -> Result<()> {
-        // find the ssh keys user and open their ssh authorized keys directory
-        let user = users::get_user_by_name(&ssh_keys_user)
-            .ok_or_else(|| format!("could not find user with username {:?}", ssh_keys_user))?;
-        let mut authorized_keys_dir = AuthorizedKeys::open(user, true, None)
-            .chain_err(|| format!("failed to open authorzied keys directory for user '{}'", ssh_keys_user))?;
+        let ssh_keys = self.ssh_keys()?;
```

Comment:
Based on my reading of the error message in the original bug report, the call to `ssh_keys()` isn't failing but actually is returning an empty key set. The `authorized_keys_dir.sync()` call is failing since it has no keys to update, and [`update-ssh-keys` won't write an empty set of keys](https://github.com/coreos/update-ssh-keys/blob/1ac025c8f203889e129c7cb8c453c9ab57725352/src/lib.rs#L258-L262), so I believe the new behavior would be correct.

Reply:
My bad, you are right. The endpoint returns a 404 and we ignore it in the client.
I just tried on an AWS t2.micro instance with no keypair and the result is:
```
$ sudo /opt/bin/coreos-metadata --cmdline --ssh-keys=core && echo OK
Jul 05 17:08:42.414 INFO Fetching http://169.254.169.254/2009-04-04/meta-data/public-keys: Attempt #1
Jul 05 17:08:42.415 INFO Fetch failed with 404: resource not found

OK
```